content/developer/howtos/rdtraining/E_unittest.rst

```diff
@@ -233,13 +233,13 @@ import the test folder/module in the ``__init__.py`` of the module.
   ├── __init__.py
   └── __manifest__.py
 
-.. note:: Some older tests are extending ``odoo.tests.common.TransactionCase``, but they are less
+.. note:: Some older tests extend ``odoo.tests.common.TransactionCase``, but they are less
   scalable. The difference is that the setup is done per test method and not per test class.
-  The data changed are rollbacked between each test in `SavepointCase` to have the same behavior as
-  in `TransactionCase`.
+  Changed data is rolled back between each test in `SavepointCase` to have the same behavior as
+  `TransactionCase`.
 
 All the tests should extend ``odoo.tests.common.SavepointCase``. You usually define a
-``setUpClass``, and the tests. After doing the `setUpClass`, you have an `env` available on the
+``setUpClass`` and the tests. After writing the `setUpClass`, you have an `env` available in the
 class and can start interacting with the ORM.
```

Comment:
This is not correct anymore.  SavepointCase is deprecated, and only TransactionCase must be used (https://github.com/odoo/odoo/pull/62031)

Reply:
Good catch. I made replace all the  SavepointCase => TransactionCase as part of the v15 commit in case someone wants to backport this commit to v14. Also deleted the note since it doesn't seem necessary any more.

While updating the technical solutions I noticed that the solution doesn't actually work out of the box and doesn't quite match the exercise instructions... so I updated the exercise part of this chapter to better match it.

Also I updated all of the advance topics to link the actual solutions repository (in the v15 changes)... I'm not sure if they're actually being used (since no one noticed some of them don't work) but hopefully this will help encourage their use.